### ISSUE TITLE
Can ZBS run the qlua and torch modes at the same time after adding the torch-7 plug-in?

### ISSUE THREAD

**Issue by TKZC0615** [pkulchenko/ZeroBraneStudio#864]
I want to debug the program in "Deep Learning with Torch: the 60-minute blitz" in ZBS .When running "itorch.image(trainset.data[100]) -- display the 100-th image in dataset" to display a picture, I have to install itorch and seem to be running only in IPython rather than in ZBS.
Therefore, I want to use "image.display" to show directly, but found that there is a mistake. After searching the Internet, I found it needed to run under qlua. What are the ways to display the pictures and debug in ZBS? 
Thank you!

**Comment by pkulchenko** [comment]
> Therefore, I want to use "image.display" to show directly, but found that there is a mistake. After searching the Internet, I found it needed to run under qlua. What are the ways to display the pictures and debug in ZBS?

Why do you think that you need to run `image.display` under qlua? `image` module depends on libimage, but I don't see any dependencies other than what's listed `image/init.lua`:

```
require 'torch'
require 'sys'
require 'xlua'
require 'dok'
require 'libimage'
```

As long as you can run this script outside of the IDE, you should be able to debug it in the IDE.

**Comment by TKZC0615** [comment]
Thank you for your reply! I can run the image test program at qlua, but in ZBS, there are the following errors:
Program completed in 3.99 seconds (pid: 7743).
/home/now/torch/install/share/lua/5.1/trepl/init.lua:389: module 'qt' not found:No LuaRocks module found for qt
	no field package.preload['qt']
	no file '/home/now/.luarocks/share/lua/5.1/qt.lua'
	no file '/home/now/.luarocks/share/lua/5.1/qt/init.lua'
	no file '/home/now/torch/install/share/lua/5.1/qt.lua'
	no file '/home/now/torch/install/share/lua/5.1/qt/init.lua'
	no file '/home/now/torch/install/install/share/lua/5.1/qt.lua'
	no file '/home/now/torch/install/install/share/lua/5.1/qt/init.lua'
	no file '/home/now/torch/install/qt.lua'
	no file '/home/now/torch/install/qt/init.lua'
	no file './qt.lua'
	no file '/home/now/torch/install/share/luajit-2.1.0-beta1/qt.lua'
	no file '/usr/local/share/lua/5.1/qt.lua'
	no file '/usr/local/share/lua/5.1/qt/init.lua'
	no file './qt/init.lua'
	no file './lua/qt.lua'
	no file './lua/qt/init.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt/qt.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt/qt/init.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt/init.lua'
	no file '/home/now/.luarocks/lib/lua/5.1/qt.so'
	no file '/home/now/torch/install/lib/lua/5.1/qt.so'
	no file '/home/now/torch/install/install/lib/lua/5.1/qt.so'
	no file '/home/now/torch/install/install/lib/lua/5.1/loadall.so'
	no file '/home/now/torch/install/install/lib/qt.so'
	no file '/home/now/torch/install/lib/lua/5.1/loadall.so'
	no file '/home/now/torch/install/lib/qt.so'
	no file '/home/now/torch/install/qt.so'
	no file './qt.so'
	no file '/usr/local/lib/lua/5.1/qt.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file '/home/now/ZeroBraneStudio/bin/linux/x64/clibs/qt.so'
	no file '/home/now/ZeroBraneStudio/bin/linux/x64/clibs/libqt.so'
stack traceback:
	[C]: in function 'error'
	/home/now/torch/install/share/lua/5.1/trepl/init.lua:389: in function 'require'
	/home/now/torch/install/share/lua/5.1/image/init.lua:1559: in function 'display'
	/home/now/桌面/untitled.lua:3: in main chunk
Debugging session completed (traced 0 instructions).

**PS: image test code is that:**
`require 'image'
l = image.lena()
image.display(l)
f = image.fabio()
image.display(f)`
Is there something wrong with some of my settings? Thank you！

**Comment by pkulchenko** [comment]
What do you see in the output window when you get this error? What process is being launched?

**Comment by TKZC0615** [comment]
I'm sorry to reply to you so late. I went home on holiday recently. As a result, a system has been reinstalled in the notebook to build the environment and restore the problem.

As I said above, that's what the output window shows.

```
Program starting as '"/home/now/ZeroBraneStudio/bin/linux/x64/lua" "/tmp/.977SVO" '.
Program 'lua' started in '/home/now/桌面/Project' (pid: 2010).
Debugging session started in '/home/now/桌面/Project/'.
tput: No value for $TERM and no -T specified
Program completed in 1.81 seconds (pid: 2010).
/home/now/torch/install/share/lua/5.1/image/init.lua:1559: module 'qt' not found:
	no field package.preload['qt']
	no file '/home/now/torch/install/share/lua/5.1/qt.lua'
	no file '/home/now/torch/install/share/lua/5.1/qt/init.lua'
	no file '/home/now/torch/share/lua/5.1/qt.lua'
	no file '/home/now/torch/share/lua/5.1/qt/init.lua'
	no file '/home/now/torch/qt.lua'
	no file '/home/now/torch/qt/init.lua'
	no file './qt.lua'
	no file '/usr/local/share/luajit-2.0.4/qt.lua'
	no file '/usr/local/share/lua/5.1/qt.lua'
	no file '/usr/local/share/lua/5.1/qt/init.lua'
	no file './qt.lua'
	no file './qt/init.lua'
	no file './lua/qt.lua'
	no file './lua/qt/init.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt/qt.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt/qt/init.lua'
	no file '/home/now/ZeroBraneStudio/lualibs/qt/init.lua'
	no file '/home/now/torch/install/lib/lua/5.1/qt.so'
	no file '/home/now/torch/install/lib/lua/5.1/loadall.so'
	no file '/home/now/torch/install/lib/qt.so'
	no file '/home/now/torch/lib/lua/5.1/qt.so'
	no file '/home/now/torch/lib/lua/5.1/loadall.so'
	no file '/home/now/torch/lib/qt.so'
	no file '/home/now/torch/qt.so'
	no file './qt.so'
	no file '/usr/local/lib/lua/5.1/qt.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file '/home/now/ZeroBraneStudio/bin/linux/x64/clibs/qt.so'
	no file '/home/now/ZeroBraneStudio/bin/linux/x64/clibs/libqt.so'
stack traceback:
	[C]: in function 'require'
	/home/now/torch/install/share/lua/5.1/image/init.lua:1559: in function 'display'
	untitled.lua:3: in main chunk
Debugging session completed (traced 0 instructions).
Can't open '': Resource temporarily unavailable
```

 **The test program for the "image" package was running at the time.**
**image test code is that:**
```
require 'image' 
l = image.lena() 
image.display(l) 
f = image.fabio() 
image.display(f)
```

**Comment by TKZC0615** [comment]
Do you have a solution to this problem? Thank you.

**Comment by TKZC0615** [comment]
In the image/init.lua file, it does have such dependencies. And in the 1559 line of this file, there are these codes.
```
   -- dependencies
   require 'qt'
   require 'qttorch'
   require 'qtwidget'
   require 'qtuiloader'
```

**Comment by pkulchenko** [comment]
> /home/now/torch/install/share/lua/5.1/image/init.lua:1559: module 'qt' not found:
> In the image/init.lua file, it does have such dependencies. And in the 1559 line of this file, there are these codes.

It does require qt, but it doesn't seem to be able to find it. So where is it located on your machine? What does `QLUA_BIN` point to in your configuraton?

**Comment by TKZC0615** [comment]
Thank you for your advice. But, No `QLUA_BIN` was found in the environment variable. Which libraries should be installed with `luarocks` before this? After running the `luarocks list` command, only `qtlua` and `qttorch` are associated with QT.

**Comment by TKZC0615** [comment]
When using ZBS to learn to debug Torch7 code, how do other people display the picture? Haven't they met these problems? I've been querying for a long time, and others use` "image"` to display pictures in the `qlua` environment, not in IDE.

**Comment by TKZC0615** [comment]
Thanks

**Comment by pkulchenko** [comment]
Yes, it's a bit puzzling, but it's possible that `qt` is not available as a standalone module and is only registered when running from qtlua. If so, this means that `require 'image'` can only be done from qtlua, which would be strange. I'll try to install qtlua myself to figure out what's going on.

**Comment by TKZC0615** [comment]
Thank you for your reply. I'll try some other way too. Thanks again！
